README.md

```diff
@@ -1,17 +1,103 @@
 * PID: Define Product ID
 
-Flashing
-=======================
-#### Versaloon
+# Flashing
+
+The microprocessor can be flashed in two different ways:
+* DFU is a simple protocol via serial port which allows programming but no debugging. On the Nitrokey hardware we expose the appropriate pins over the USB connector but it's not USB! Details are described in the next chapter.
+* SWD is a STM-specific protocol and similar to JTAG allowing programming and debugging. Working adapters are Versaloon or any of the ST-Link V2 (clones). Under Linux you could give a patched OpenOCD a try but in the past it has been very troublesome. This approach requires to solder wires to the contact pads or to use an adapter with pogo pins and some kind of mounting (recommended).
+
+## STM32flash and DFU Mode
+### Requirements
+
+* Download the .hex file you want to flash e.g. look at the [releases section](https://github.com/Nitrokey/nitrokey-pro-firmware/releases) or build it yourself (see above).
+* You may use [STM32 Flash Loader Demonstrator](http://www.st.com/content/st_com/en/products/development-tools/software-development-tools/stm32-software-development-tools/stm32-programmers/flasher-stm32.html) (Windows only) or the open source command line tool [stm32flash](http://stm32flash.sourceforge.net). *Note: the terminal commands below are based on the command line tool.*
+* If your computer doesn't has a RS232 port (most modern laptops don't have it) you would need a USB-to-RS232/TTL adapter. [Sparkfun BOB-00718](https://www.sparkfun.com/products/718) should work (untested) and you can find even cheaper adapters online. Previously we built [our own adapter](https://www.assembla.com/spaces/cryptostick/wiki/Crypto_Stick_Programmer) which hardware layout you can [download](https://www.assembla.com/spaces/cryptostick/subversion/source/HEAD/trunk/hardware/programmer%20for%20v1.2).
+* You would need a simple USB adapter to bridge Nitrokey's USB plug to the USB-to-RS232 adapter.
+
+Your adapter should consist of a USB socket which four pins are connected to your serial/TTL connector. The pinout is as follows.
+
+```
+Nitrokey USB Plug <-> Serial/TTL adapter
+
+Pin 1, VCC <-> VCC
+Pin 2, D-  <-> TX
+Pin 3, D+  <-> RX
+Pin 4, GND <-> GND
+
+This diagram represents the pinout of the USB socket which you are going to solder:
+
+  ################### 
+  #                 # 
+  # ############### # 
+  #                 # 
+  #                 # 
+  ################### 
+     #   #   #   #   
+     #   #   #   #    
+
+     1   2   3   4
+```
+
+The following picture shows the adapter/USB-to-TTL connection.
+![USB-to-TTL adapter and USB socket](adapter.jpg?raw=true)
+
+To flash the firmware you need to bridge the two contact holes and only then connect (and power) the PCB to your adapter. The bridge triggers the hardware to boot into DFU mode. You can use a jumper with 2.0 mm pitch or just prepare/solder a wire. The following picture shows a bridge for the Nitrokey.
```

Comment:
@jans23 Would that still be true with Nitrokey Pro 2?

Reply:
Note about hardware revision added